dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java

```diff
@@ -304,67 +311,73 @@ protected String getData(RyaType ryaType) {
         final Long ttl = ryaQuery.getTtl();
-        Long currentTime = ryaQuery.getCurrentTime();
-        Long maxResults = ryaQuery.getMaxResults();
-        Integer batchSize = ryaQuery.getBatchSize();
-        Integer numQueryThreads = ryaQuery.getNumQueryThreads();
-        String regexSubject = ryaQuery.getRegexSubject();
-        String regexPredicate = ryaQuery.getRegexPredicate();
-        String regexObject = ryaQuery.getRegexObject();
-        TableLayoutStrategy tableLayoutStrategy = configuration.getTableLayoutStrategy();
-        int maxRanges = ryaQuery.getMaxRanges();
+        final Long currentTime = ryaQuery.getCurrentTime();
+        final Long maxResults = ryaQuery.getMaxResults();
+        final Integer batchSize = ryaQuery.getBatchSize();
+        final Integer numQueryThreads = ryaQuery.getNumQueryThreads();
+        final String regexSubject = ryaQuery.getRegexSubject();
+        final String regexPredicate = ryaQuery.getRegexPredicate();
+        final String regexObject = ryaQuery.getRegexObject();
+        final TableLayoutStrategy tableLayoutStrategy = configuration.getTableLayoutStrategy();
+        final int maxRanges = ryaQuery.getMaxRanges();
 
         // TODO: cannot span multiple tables here
         try {
-            Collection<Range> ranges = new HashSet<Range>();
+            final Collection<Range> ranges = new HashSet<Range>();
+
             TABLE_LAYOUT layout = null;
             RyaURI context = null;
+
             TriplePatternStrategy strategy = null;
-            for (RyaStatement stmt : stmts) {
+            for (final RyaStatement stmt : stmts) {
                 context = stmt.getContext(); // TODO: This will be overwritten
                 strategy = ryaContext.retrieveStrategy(stmt);
                 if (strategy == null) {
                     throw new IllegalArgumentException("TriplePattern[" + stmt + "] not supported");
                 }
 
-                Map.Entry<RdfCloudTripleStoreConstants.TABLE_LAYOUT, ByteRange> entry = strategy.defineRange(stmt.getSubject(),
-                        stmt.getPredicate(), stmt.getObject(), stmt.getContext(), null);
+                final Map.Entry<RdfCloudTripleStoreConstants.TABLE_LAYOUT, ByteRange> entry = strategy.defineRange(stmt.getSubject(), stmt.getPredicate(), stmt.getObject(), stmt.getContext(), null);
 
                 // use range to set scanner
                 // populate scanner based on authorizations, ttl
                 layout = entry.getKey();
-                ByteRange byteRange = entry.getValue();
+                final ByteRange byteRange = entry.getValue();
+
                 Range range = new Range(new Text(byteRange.getStart()), new Text(byteRange.getEnd()));
+                if ((stmt.getSubject() != null) && (stmt.getPredicate() != null) && (stmt.getObject() != null)) {
+                    range = Range.exact(createText(byteRange.getStart(), stmt.getObject()));
```

Comment:
does `byteRange.getStart()` ever have the type appended to it at this point?  (if so, then this won't work)